scripts/ci/_utils.sh

```diff
@@ -910,15 +927,39 @@ function build_image_on_ci() {
     # Cleanup docker installation. It should be empty in CI but let's not risk
     docker system prune --all --force
     rm -rf "${BUILD_CACHE_DIR}"
+    mkdir -pv "${BUILD_CACHE_DIR}"
 
-    if [[ ${TRAVIS_JOB_NAME:=""} == "Tests"* ]]; then
-        rebuild_ci_image_if_needed
+    CHANGED_FILE_NAMES=$(git diff --name-only "${TRAVIS_BRANCH:="HEAD"}...HEAD")
+    echo
+    echo "Changed file names in this commit"
+    echo "$CHANGED_FILE_NAMES"
+    echo
+
+    if [[ ${TRAVIS_JOB_NAME:=""} == "Tests"*"kubernetes"* ]]; then
+        match_files_regexp 'airflow/kubernetes/.*\.py' 'tests/kubernetes/.*\.py'
```

Comment:
I've had a few cases where bugs in my PR in the webserver code were only caught by the kube tests btw.

Reply:
I also changed it so that all tests are always run in CRON job run daily. I do expect some false positives to slip through.

For now I think it is best to include airflow/www/*.py in kubernetes and airflow/www/static/*.js  in general (not only .py) to both regexps. That should handle big class of potential false positives. 

I think it's a good safety net eventually. 

Possibly if we find there are other dependencies, we can add them after we see they fail in the CRON job.

Reply:
That sounds like a good compromise (until we add actual "integration" tests for Airflow webserver. It's on Astronomer's pipeline, so we'll get around to it at some point.)